instrumentation/net/http/httptrace/otelhttptrace/test/clienttrace_test.go

```diff
@@ -42,107 +47,153 @@ func getSpansFromRecorder(sr *tracetest.SpanRecorder, name string) []trace.ReadO
+			}
+			_ = res.Body.Close()
 
-		res, err := client.Do(req)
+			return nil
+		}(context.Background())
 		if err != nil {
-			t.Fatalf("Request failed: %s", err.Error())
+			panic("unexpected error in http request: " + err.Error())
 		}
-		_ = res.Body.Close()
 
-		return nil
-	}(context.Background())
-	if err != nil {
-		panic("unexpected error in http request: " + err.Error())
-	}
+		type tc struct {
+			name       string
+			attributes []attribute.KeyValue
+			parent     string
+		}
 
-	testLen := []struct {
-		name       string
-		attributes []attribute.KeyValue
-		parent     string
-	}{
-		{
-			name: "http.connect",
-			attributes: []attribute.KeyValue{
-				attribute.Key("http.conn.done.addr").String(address.String()),
-				attribute.Key("http.conn.done.network").String("tcp"),
-				attribute.Key("http.conn.start.network").String("tcp"),
-				attribute.Key("http.remote").String(address.String()),
+		testLen := []tc{
+			{
+				name: "http.connect",
+				attributes: []attribute.KeyValue{
+					attribute.Key("http.conn.done.addr").String(ts.Listener.Addr().String()),
+					attribute.Key("http.conn.done.network").String("tcp"),
+					attribute.Key("http.conn.start.network").String("tcp"),
+					attribute.Key("http.remote").String(ts.Listener.Addr().String()),
+				},
+				parent: "http.getconn",
 			},
-			parent: "http.getconn",
-		},
-		{
-			name: "http.getconn",
-			attributes: []attribute.KeyValue{
-				attribute.Key("http.remote").String(address.String()),
-				attribute.Key("net.host.name").String(address.String()),
-				attribute.Key("http.conn.reused").Bool(false),
-				attribute.Key("http.conn.wasidle").Bool(false),
+			{
+				name: "http.getconn",
+				attributes: []attribute.KeyValue{
+					attribute.Key("http.remote").String(ts.Listener.Addr().String()),
+					attribute.Key("net.host.name").String(ts.Listener.Addr().String()),
+					attribute.Key("http.conn.reused").Bool(false),
+					attribute.Key("http.conn.wasidle").Bool(false),
+				},
+				parent: "test",
+			},
+			{
+				name:   "http.receive",
+				parent: "test",
+			},
+			{
+				name:   "http.headers",
+				parent: "test",
+			},
+			{
+				name:   "http.send",
+				parent: "test",
+			},
+			{
+				name: "test",
 			},
-			parent: "test",
-		},
-		{
-			name:   "http.receive",
-			parent: "test",
-		},
-		{
-			name:   "http.headers",
-			parent: "test",
-		},
-		{
-			name:   "http.send",
-			parent: "test",
-		},
-		{
-			name: "test",
-		},
-	}
-	for _, tl := range testLen {
-		span, ok := getSpanFromRecorder(sr, tl.name)
-		if !assert.True(t, ok) {
-			continue
 		}
 
-		if tl.parent != "" {
-			parent, ok := getSpanFromRecorder(sr, tl.parent)
-			if assert.True(t, ok) {
-				assert.Equal(t, span.Parent().SpanID(), parent.SpanContext().SpanID())
-			}
+		u, err := url.Parse(ts.URL)
+		if err != nil {
+			panic("unexpected error in parsing httptest server URL: " + err.Error())
+		}
+		// http.tls only exists on HTTPS connections.
+		if u.Scheme == "https" {
```

Comment:
There shouldn't be conditions like this in a subtest loop. These should be part of the attributes defined in the loop definition.